### PR TITLE
Migrate ExoPlayer to SDK and decouple from webapp

### PR DESCRIPTION
**Pull request by Maxr1998**
Now uses the new Kotlin SDK to resolve media metadata, create the device profile, initialise playback and report playback progress/events.
Supports playlists.

Smaller changes:
- Media sessions now always show a notification, not only when in background
- The notification has a skip to previous/next button when playing a playlist now instead of seek
- Playback progress is only reported regularly when media is playing
- The fullscreen-switcher on portrait videos now has the correct initial state
- Components/factories for ExoPlayer are injected with Koin

### CODE REVIEW

File: app/src/main/assets/native/ExoPlayerPlugin.js [hunk 1]
```diff
@@ -1,73 +1,55 @@
 export class ExoPlayerPlugin {
     constructor({ appSettings, events, playbackManager, loading }) {
-        window['ExoPlayer'] = this;
+        window["ExoPlayer"] = this;
```
Comment (nielsvanvelzen) [inline app/src/main/assets/native/ExoPlayerPlugin.js:3]: Why did you change the quotes? It's pretty much a standard to use single quotes in JavaScript

File: app/src/main/assets/native/ExoPlayerPlugin.js [hunk 2]
```diff
@@ -1,73 +1,55 @@
...
-
-    currentSrc() {
-        return this._currentSrc;
+        this._nativePlayer = window["NativePlayer"];
     }
 
     async play(options) {
-        this.prepareAudioTracksCapabilities(options);
+        // Sanitize input
+        options.ids = options.items.map((item) => item.Id);
```
Comment (nielsvanvelzen) [inline app/src/main/assets/native/ExoPlayerPlugin.js:27]:
```suggestion
        options.ids = options.items.map(item => item.Id);
```

File: app/src/main/assets/native/ExoPlayerPlugin.js [hunk 3]
```diff
@@ -78,308 +60,122 @@ export class ExoPlayerPlugin {
...
-        this.subtitleStreamIndex = innerIndex;
-    }
+    setSubtitleStreamIndex(index) {}
 
-    async changeAudioStream(index) {
-        var innerIndex = Number(index);
-        this.playbackManager.setAudioStreamIndex(innerIndex);
-        this.audioStreamIndex = innerIndex;
-    }
+    async changeAudioStream(index) {}
```
Comment (nielsvanvelzen) [inline app/src/main/assets/native/ExoPlayerPlugin.js:144]: Some previously implemented functions, like this one, are now empty. Why?

They can still be called when using remote control for example.

File: app/src/main/java/org/jellyfin/mobile/player/PlayerNotificationHelper.kt [hunk 4]
```diff
@@ -84,17 +83,25 @@ class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinCom
                 }
                 setSmallIcon(R.drawable.ic_notification)
                 mediaIcon?.let(::setLargeIcon)
-                setContentTitle(mediaSource.title)
-                mediaSource.artists?.let(::setContentText)
+                setContentTitle(mediaSource.name)
+                //mediaSource.sourceInfo.artists?.let(::setContentText) // TODO
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/player/PlayerNotificationHelper.kt:87]: ?

File: app/src/main/java/org/jellyfin/mobile/player/source/JellyfinMediaSource2.kt [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import org.jellyfin.mobile.utils.Constants
+import org.jellyfin.sdk.model.api.MediaSourceInfo
+import org.jellyfin.sdk.model.api.MediaStream
+import org.jellyfin.sdk.model.api.MediaStreamType
+import org.jellyfin.sdk.model.api.PlayMethod
+import org.jellyfin.sdk.model.serializer.toUUIDOrNull
+import java.util.*
+import kotlin.collections.ArrayList
+
+class JellyfinMediaSource2(
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/player/source/JellyfinMediaSource2.kt:12]: Not a fan of this name tbh

File: app/src/main/assets/native/ExoPlayerPlugin.js (hunk 1 above)
Comment (Maxr1998) [reply app/src/main/assets/native/ExoPlayerPlugin.js:3]: That was my VSCode :grimacing:

File: app/src/main/assets/native/ExoPlayerPlugin.js (hunk 3 above)
Comment (Maxr1998) [reply app/src/main/assets/native/ExoPlayerPlugin.js:144]: Their previous implementation wasn't really handled either. So, they didn't work, and definitely won't in the new implementation. Still have to look into how to handle this with the SDK.

File: app/src/main/java/org/jellyfin/mobile/player/PlayerNotificationHelper.kt (hunk 4 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/player/PlayerNotificationHelper.kt:87]: I don't have access to the item anymore. Should probably request it through the API when fetching the media source info.

File: app/src/main/java/org/jellyfin/mobile/player/source/JellyfinMediaSource2.kt (hunk 5 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/player/source/JellyfinMediaSource2.kt:12]: Me neither. I'd prefer to get rid of the old one, but ExternalPlayer uses it xD

File: app/src/main/assets/native/ExoPlayerPlugin.js (hunk 3 above)
Comment (nielsvanvelzen) [reply app/src/main/assets/native/ExoPlayerPlugin.js:144]: Something for the fully native app then :p

File: app/src/main/java/org/jellyfin/mobile/player/source/JellyfinMediaSource2.kt (hunk 5 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/mobile/player/source/JellyfinMediaSource2.kt:12]: Is it possible to migrate the externalplayer to this one?

File: app/src/main/java/org/jellyfin/mobile/player/PlayerNotificationHelper.kt (hunk 4 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/mobile/player/PlayerNotificationHelper.kt:87]: Well please do that or remove this line then :p

File: app/src/main/java/org/jellyfin/mobile/player/source/JellyfinMediaSource2.kt (hunk 5 above)
Comment (Maxr1998) [reply app/src/main/java/org/jellyfin/mobile/player/source/JellyfinMediaSource2.kt:12]: Need to check that, but could be possible.

File: app/src/main/java/org/jellyfin/mobile/player/PlayOptions.kt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+data class PlayOptions(
+    val mediaSourceId: UUID?,
+    val ids: List<UUID>,
+    val startIndex: Int,
+    val startPositionTicks: Long?,
+    val audioStreamIndex: Int?,
+    val subtitleStreamIndex: Int?,
+) : Parcelable {
+
+    companion object {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/player/PlayOptions.kt:24]:
```suggestion
    companion object {
```

**Review by nielsvanvelzen — APPROVED** [review]

**Review by nielsvanvelzen — APPROVED** [review]
